common/src/main/java/com/wynntils/models/skillpoint/SkillPointModel.java

```diff
@@ -0,0 +1,386 @@
+                negatives.put(Skill.values()[i], difference);
+            }
+        }
+
+        AtomicBoolean confirmationCompleted = new AtomicBoolean(false);
+        negatives.forEach((skill, difference) -> {
+            int difference5s = Math.abs(difference) / 5;
+            int difference1s = Math.abs(difference) % 5;
+
+            for (int i = 0; i < difference5s; i++) {
+                ContainerUtils.clickOnSlot(
+                        SKILL_POINT_TOTAL_SLOTS[skill.ordinal()],
+                        containerContent.containerId(),
+                        GLFW.GLFW_MOUSE_BUTTON_RIGHT,
+                        containerContent.items(),
+                        true);
+                if (!confirmationCompleted.getAndSet(true)) {
+                    // confirmation required, force loop to repeat this iteration
+                    i--;
+                }
+            }
+            for (int i = 0; i < difference1s; i++) {
+                ContainerUtils.clickOnSlot(
+                        SKILL_POINT_TOTAL_SLOTS[skill.ordinal()],
+                        containerContent.containerId(),
+                        GLFW.GLFW_MOUSE_BUTTON_RIGHT,
+                        containerContent.items(),
+                        false);
+                if (!confirmationCompleted.getAndSet(true)) {
+                    // needs to exist in both loops in case of 1s only
+                    i--;
+                }
+            }
+        });
+
+        positives.forEach((skill, difference) -> {
+            int difference5s = difference / 5;
+            int difference1s = difference % 5;
+
+            for (int i = 0; i < difference5s; i++) {
+                ContainerUtils.clickOnSlot(
+                        SKILL_POINT_TOTAL_SLOTS[skill.ordinal()],
+                        containerContent.containerId(),
+                        GLFW.GLFW_MOUSE_BUTTON_LEFT,
+                        containerContent.items(),
+                        true);
+            }
+            for (int i = 0; i < difference1s; i++) {
+                ContainerUtils.clickOnSlot(
+                        SKILL_POINT_TOTAL_SLOTS[skill.ordinal()],
+                        containerContent.containerId(),
+                        GLFW.GLFW_MOUSE_BUTTON_LEFT,
+                        containerContent.items(),
+                        false);
+            }
+        });
+
+        // Server needs 2 ticks, give one extra to be safe
+        Managers.TickScheduler.scheduleLater(this::populateSkillPoints, 3);
+    }
+
+    public void deleteLoadout(String name) {
+        skillPointLoadouts.get().remove(name);
+    }
+
+    /**
+     * Closes any open containers (but not the screen shown) then queries compass (and tome menus) depending on if
+     * tomes have been unlocked.
+     */
+    public void populateSkillPoints() {
+        McUtils.closeBackgroundContainer();
+
+        Managers.TickScheduler.scheduleNextTick(() -> {
+            assignedSkillPoints.clear();
```

Comment:
I would rather you just set it to a new instance, which is usually safer in case something holds a reference to the object, we won't crash with CCME.